src/poke_env/environment/abstract_battle.py

```diff
@@ -1221,6 +1228,18 @@ def teampreview(self) -> bool:
         """
         return self._teampreview
 
+    @property
+    def teampreview_team(self) -> Set[Pokemon]:
```

Comment:
what is currently happening for this in teampreview? is our team not accessible in `self.team`? if so, why not store it in teampreview_team instead?

Reply:
I think there are two philosophical reasons I think it would make sense to store teampreview_team. I say "philosophical" because I could see people falling on either side and being well-thought through.

Philosophical idea: The battle object is supposed to capture the state of the battle, as it pertains to making optimal decisions. This means it should capture not only what we know, but also what the opponent knows. Thus, we need to store what the opponent thinks we might have for when we pass the battle state into our Deep Learning network of choice (you can also make the case that this should be done via player, which would also be reasonable).

Reply:
i see - this is reasonnable